src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/DeploymentOptions.cs

```diff
@@ -297,5 +303,19 @@ public void SetOptions(DacDeployOptions options)
                 }
             }
         }
+
+        public static DeploymentOptions GetDefaultSchemaCompareOptions()
```

Comment:
I like this idea of having clear apis, few things to look closely 

1. why are we having deply option != SC option (is it cause its different today in SSDT?)
2. are we having SC option in ADS == SC option in SSDT? I think that should be the case to start with yeah?
3. Depoly option for project vs for dacpac ext - will thhey be same?

4. more on API - We do have an this api for SC - right? and do we plan with this change to use the new api for deploy/extract too?

Reply:
This change as is shouldn't change anything so far as differences (or similarities) between ADS SC, ADS Publish, or SSDT.  All it's currently doing is moving* this logic from a previous bugfix from ADS to Tools Service where it makes more sense: https://github.com/microsoft/azuredatastudio/blob/7bb4d000730f6c24a9cb421f641c97fbda6f35b2/extensions/sql-database-projects/src/dialogs/publishDatabaseDialog.ts#L222

* technically it isn't moving it yet, it's duplicating the logic so the ADS side is a no-op, and I'll remove that logic from ADS once we update Tools Service in that branch.

Reply:
makes sense

Reply:
1. We hit a bug in Publish a while back where DatabaseScopedCredentials needed to be included by default, but weren't because we were using the SchemaCompare defaults.  I _think_ @kisantia said that she'd copied the SC defaults from SSDT, so we shouldn't change, so I added it just for the publish flow.  I think this was an askmode bug, so we did the fix just within the extension where we wouldn't delay the release rather than having to port a new version of Tools Service.

2.  I think so?

3. This change shouldn't affect the Dacpac Publish flow; if it was already different from Project Publish, it'll continue to be.  Having said that, I think they _should_ be the same...  See thoughts below.

4. Which API for SC/Deploy/Extract?  Probably also related to thoughts below.

Thoughts:
I think we should audit what the defaults currently are for the different ways to trigger DacFx ops (SSDT/ADS/SqlPacakge x Schema Compare/Publish) so we can see where the differences are, and decide whether they're intentional or are drift due to having defaults getting set in different places.  Then we can consolidate the logic for determining defaults for different operations and eliminate the drift.